unsafe/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -380,7 +405,13 @@ public void putNewKey(
       bitset.set(pos);
 
       // If there's not enough space in the current page, allocate a new page:
-      if (currentDataPage == null || PAGE_SIZE_BYTES - pageCursor < requiredSize) {
+      if (currentDataPage == null || PAGE_SIZE_BYTES - pageCursor - 8 < requiredSize) {
+        if (currentDataPage != null) {
```

Comment:
Well, I guess we'll still end up allocating a new block.  I'll add a comment to make this more clear.